### ISSUE TITLE
CategoricalIndex.reindex raises on duplicate indexer

### ISSUE THREAD

**Issue by adbull** [pandas-dev/pandas#25459]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

# works fine
x = pd.Series([0])
x.reindex([0,0])

# ValueError: cannot reindex with a non-unique indexer
x.index = x.index.astype('category')
x.reindex([0,0])
```

#### Problem description

When a series or frame has a categorical index, it cannot be reindexed using an indexer that has duplicate values. This is inconsistent with other indexes, and makes categorical indexes less useful.

#### Expected Output

No error.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Linux
OS-release: 4.19.15-200.fc28.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

pandas: 0.24.1
pytest: 4.3.0
pip: 19.0.3
setuptools: 40.8.0
Cython: 0.29.5
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 4.2.1
sphinx: None
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.2
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.3.1
bs4: 4.7.1
html5lib: None
sqlalchemy: 1.2.18
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

**Comment by gfyoung** [comment]
cc @jreback @toobaz

**Comment by batterseapower** [comment]
This was broken by #23963 which was trying to fix #21809. The problem is that this line should be checking `self.is_unique` instead of `target.is_unique`: https://github.com/pandas-dev/pandas/blame/master/pandas/core/indexes/category.py#L602
